### ISSUE TITLE
perf: vectorize cos_sim calculation using tensor batching in check_duplicate

### ISSUE THREAD

**Issue by suhaniiz** [riteshbonthalakoti/HELPDESK.AI#2655]
### Description
In `DuplicateService.check_duplicate()`, the application currently uses a Python `for` loop to sequentially calculate cosine similarity against every single stored ticket execution block:

```python
# CURRENT LINEAR LOOP DETECTED
best_score = 0.0
best_id = None

for ticket_id, stored_emb, _ in self._tickets:
    score = util.cos_sim(query_embedding, stored_emb).item()
    if score > best_score:
        best_score = score
        best_id = ticket_id
```
Python loops defeat the purpose of utilizing PyTorch and vectorization. Running individual scalar computations causes a linear execution slowdown processing latency) as the in-memory cache grows, which will degrade performance under high-throughput request loads.

Proposed Solution
Refactor Internal Data Structures: Instead of a list of tuples, maintain parallel arrays for structural components (list[str] for IDs) and stack vectors into a multi-dimensional matrix tensor (torch.Tensor).

Dynamic Matrix Appending: When calling add_ticket(), use torch.cat() to append the new tensor array row vertically along dim=0.

Vectorized Matrix Multiplication: In check_duplicate(), pass the query_embedding and the consolidated embeddings_tensor to util.cos_sim() once. Extract the highest matching vector score index instantly using torch.argmax().

**Comment by riteshbonthalakoti** [comment]
Hi @suhaniiz! Welcome aboard, I have assigned this issue to you. Make sure to sign up under the company **Ritesh PVT Limited** when testing.

Also, make sure you complete the mandatory onboarding steps so your points register correctly:

⚠️ **MANDATORY GSSOC ONBOARDING STEPS**:
Before your PR points are finalized on the leaderboard, you **MUST** complete these steps:
1. ⭐ **Star this repository**: https://github.com/ritesh-1918/HELPDESK.AI (Mandatory)
2. 👤 **Follow the Project Admin**: https://github.com/ritesh-1918 (Mandatory - manual step)
3. 💼 **Connect on LinkedIn**: https://www.linkedin.com/in/ritesh1908/ (Mandatory)


Happy coding! 🚀
